api/script/views/authenticate.ejs

```diff
@@ -85,11 +85,15 @@
     <div class="container">
         Please select an authentication provider for your CodePush account:<br />
         <br />
-        <a href="/auth/<%=action%>/github"><img class="github" src="images/GitHub.png"> GitHub</a>
-        <span class="spacer"/>
-        <a href="/auth/<%=action%>/microsoft"><img class="microsoft" src="images/windows10.png"> Microsoft (Personal)</a>
-        <span class="spacer"/>
-        <a href="/auth/<%=action%>/azure-ad"><img class="microsoft" src="images/windows10.png"> Microsoft (Work)</a><br />
+        <% if (isGitHubAuthenticationEnabled) { %>
```

Comment:
We don't want to hide authentication if it is not available.

Reply:
> We don't want to hide authentication if it is not available.

So, a broken button is better ? 🙃